lib/memcachir.ex

```diff
@@ -84,8 +84,10 @@ defmodule Memcachir do
 
   """
   def set(key, value, opts \\ []) do
+    {retry, opts} = Keyword.pop(opts, :retry, false)
```

Comment:
Since the set method already has an options argument I leveraged that to pass a retry parameter. Since opts are passed further down to the memcachex lib, we pop :retry since only this lib cares about it.